### PR TITLE
DEV: Add system tests

### PR DESCRIPTION
**Pull request by tgxworld**
Why does this commit do?

This commit improves the test coverage for the theme by adding system tests which covers the basic functionality of the theme.

### CODE REVIEW

File: javascripts/discourse/components/custom-footer.js [hunk 1]
```diff
@@ -32,13 +32,15 @@ export default class extends Component {
...
       const text = fragments[0];
+      const title = fragments[1];
       const dataName = dasherize(text);
       const childLinks = this.linkArray.filter(
         (link) => link.parent === parentFor
       );
 
       return {
         text,
+        title,
```
Comment (tgxworld) [inline javascripts/discourse/components/custom-footer.js:43]: Caught a bug here while working on the system test.

File: spec/system/footer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        "a.social-link[data-easyfooter-social-link='twitter'][title='show some love on Twitter'][href='#'][target='_blank'] .d-icon-fab-twitter",
+      )
+
+      expect(page).to have_css(
+        "a.social-link[data-easyfooter-social-link='youtube'][title='Check out our latest videos on Youtube'][href='#'][target='_blank'] .d-icon-fab-youtube",
+      )
+    end
+  end
+
+  it "should not display the footer to anon users when `Show_footer_on_login_required_page` is false" do
```
Comment (martin-brennan) [inline spec/system/footer_spec.rb:77]: I guess it's already a legacy thing but it's so weird the theme setting is capitalized :grimacing:

File: spec/system/footer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+RSpec.describe "Footer", system: true do
+  let!(:theme) { upload_theme_component }
+
+  it "should display the footer" do
+    theme.update_setting(:Link_sections, "Section 1, Section 1 title|Section 2, Section 2 title")
```
Comment (martin-brennan) [inline spec/system/footer_spec.rb:7]: Oh maybe the capitalization is a theme setting thing generally? Seems strange to me, but not related to this PR

**Review by martin-brennan — APPROVED** [review]

File: spec/system/footer_spec.rb (hunk 2 above)
Comment (tgxworld) [reply spec/system/footer_spec.rb:77]: Yup this is something we will fix soon when we have the ability to migrate theme settings.

**Comment by tgxworld** [comment]
@martin-brennan Thank you for the review :+1:

**Pull request merged**